pkg/podman/version.go

```diff
@@ -0,0 +1,38 @@
+package podman
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+)
+
+type Version struct {
+	APIVersion string
+	Version    string
+	GoVersion  string
+	GitCommit  string
+	BuiltTime  string
+	Built      int64
+	OsArch     string
+	Os         string
+}
+
+type SystemVersionReport struct {
+	Client *Version `json:",omitempty"`
+}
+
+func (o *PodmanCli) Version() (SystemVersionReport, error) {
+	out, err := exec.Command(o.podmanCmd, "version", "--format", "json").Output()
```

Comment:
Wondering if we shouldn't `klog` the actual command that will be launched (here and everywhere we are running `podman`).
I didn't remember I had set `export PODMAN_CMD=docker` on my system some time ago. And the error message (even with increased verbosity) was not really helpful in understanding the error.

```
$ ODO_EXPERIMENTAL_MODE=true odo dev --run-on=podman --debug --no-watch
Experimental mode enabled for flag: --run-on. Use at your own risk. More details on https://odo.dev/docs/user-guides/advanced/experimental-mode
 ✗  unable to access podman. Do you have podman client installed?
```